ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/tokenize/SparseTokenizerModel.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ml.engine.algorithms.tokenize;
+
+import ai.djl.MalformedModelException;
+import ai.djl.huggingface.tokenizers.Encoding;
+import ai.djl.huggingface.tokenizers.HuggingFaceTokenizer;
+import ai.djl.inference.Predictor;
+import ai.djl.modality.Input;
+import ai.djl.modality.Output;
+import ai.djl.repository.zoo.ModelNotFoundException;
+import ai.djl.repository.zoo.ZooModel;
+import ai.djl.translate.TranslateException;
+import ai.djl.translate.Translator;
+import ai.djl.translate.TranslatorFactory;
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import lombok.extern.log4j.Log4j2;
+import org.opensearch.ml.common.FunctionName;
+import org.opensearch.ml.common.dataset.MLInputDataset;
+import org.opensearch.ml.common.dataset.TextDocsInputDataSet;
+import org.opensearch.ml.common.input.MLInput;
+import org.opensearch.ml.common.model.MLModelConfig;
+import org.opensearch.ml.common.output.model.ModelResultFilter;
+import org.opensearch.ml.common.output.model.ModelTensor;
+import org.opensearch.ml.common.output.model.ModelTensorOutput;
+import org.opensearch.ml.common.output.model.ModelTensors;
+import org.opensearch.ml.engine.algorithms.DLModel;
+import org.opensearch.ml.engine.annotation.Function;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.lang.reflect.Type;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static org.opensearch.ml.common.utils.StringUtils.gson;
+import static org.opensearch.ml.common.CommonValue.ML_MAP_RESPONSE_KEY;
+import static org.opensearch.ml.engine.utils.FileUtils.deleteFileQuietly;
+
+@Log4j2
+@Function(FunctionName.SPARSE_TOKENIZE)
+public class SparseTokenizerModel extends DLModel {
```

Comment:
I am a little bit confused about this model class. The DLModel is for DJL-based deep learning model implementation. I don't see an underlying model inference instead of a static idf dict usage. There is no term expansion also. Why don't we have a sparse model here? If we only need a bert-tokenizer and an idf dict, no need to implement a DLModel interface. Just a normal OpenSearch tokenizer is OK as no model execution needed.

Reply:
And it is also hard to maintain the consistency between the sparse model used in ingestion and the idf json file used in query.

Reply:
The tokenize model will also need to combine the divided zip file and load from that. The logic is same as the DL model. That's the first reason why I extend DLModel here. And as we discussed before, tokenize model should be a model can be deployed and registered, and customer needs to maintain the consistency between tokenizer and model, also maintain the consistency between the idf and tokenizer inside the tokenizer zip file by themselves. So we don't reuse TextEmbeddingSparseEncodingModel here or add a field in request body.

Reply:
Thank you for your detailed explanation. I also synced up with Yaliang and we think it would be better to have some comments here to describe 
1. how to generate the idf.json
2. users need to maintain the consistency of the sparse tokenizer and the sparse encoding model on their own.
3. the tradeoff between search latency and the ranking/scoring accuracy.

Reply:
Will add this comment to tokenizerModel. Thanks.